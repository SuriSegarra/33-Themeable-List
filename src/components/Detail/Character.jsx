import React from 'react';
import PropTypes from 'prop-types';

const Character = ({ image, name, species, status, origin }) => (
  <figure>
    <img src={image} alt={name}/>
    <figcaption>
      <h1>{name}</h1>
      <p>Specie:{species}</p>
      <p>Status:{status}</p>
      <p>origin:{origin}</p>
    </figcaption>
  </figure>
);

Character.propTypes = {
  image: PropTypes.string.isRequired,
  name: PropTypes.string.isRequired,
  species: PropTypes.string.isRequired,
  status: PropTypes.string.isRequired,
  origin: PropTypes.string.isRequired
};

export default Character;
