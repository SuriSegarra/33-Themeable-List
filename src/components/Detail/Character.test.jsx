import React from 'react';
import { shallow } from 'enzyme';
import Character from './Character';

describe('Character component', () => {
  it('renders Character', () => {
    const wrapper = shallow(<Character 
      name= 'Rick Sanchez'
      image='https://rickandmortyapi.com/api/character/avatar/1.jpeg'
      species='human'
      status='alive'
      origin='earth'/>);
    expect(wrapper).toMatchSnapshot();
  });
});
