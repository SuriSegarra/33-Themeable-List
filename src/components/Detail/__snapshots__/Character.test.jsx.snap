// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Character component renders Character 1`] = `
<figure>
  <img
    alt="Rick Sanchez"
    src="https://rickandmortyapi.com/api/character/avatar/1.jpeg"
  />
  <figcaption>
    <h1>
      Rick Sanchez
    </h1>
    <p>
      Specie:
      human
    </p>
    <p>
      Status:
      alive
    </p>
    <p>
      origin:
      earth
    </p>
  </figcaption>
</figure>
`;
